android/src/main/kotlin/studio/midoridesign/gal/GalPlugin.kt

```diff
@@ -0,0 +1,304 @@
+        val status: Int = ContextCompat.checkSelfPermission(context, PERMISSION)
+        return status == PackageManager.PERMISSION_GRANTED
+    }
+
+    private fun requestAccess() {
+        activity?.let {
+            ActivityCompat.requestPermissions(
+                it, arrayOf(PERMISSION),
+                PERMISSION_REQUEST_CODE
+            )
+        }
+    }
+
+    private fun sendError(
+        errorCode: String, message: String, stackTrace: Array<StackTraceElement>,
+        result: MethodChannel.Result
+    ) {
+        val trace = StringBuilder()
+        for (st in stackTrace) {
+            trace.append(st.toString())
+            trace.append("\n")
+        }
+        Handler(Looper.getMainLooper())
+            .post { result.error(errorCode, message, trace.toString()) }
+    }
+
+    private fun handleError(e: Exception, result: MethodChannel.Result) {
+        val errorCode: String =
+            if (e is SecurityException || e.toString().contains("Permission denied")) {
+                "ACCESS_DENIED"
+            } else if (e is FileNotFoundException) {
+                "NOT_SUPPORTED_FORMAT"
+            } else if (e is IOException && e.toString().contains("No space left on device")) {
+                "NOT_ENOUGH_SPACE"
+            } else {
+                "UNEXPECTED"
+            }
+        sendError(errorCode, e.toString(), e.stackTrace, result)
+    }
+
+    override fun onAttachedToActivity(activityPluginBinding: ActivityPluginBinding) {
+        activity = activityPluginBinding.activity
+        activityPluginBinding.addRequestPermissionsResultListener(this)
+    }
+
+    override fun onDetachedFromActivity() {
+        activity = null
+    }
+
+    override fun onReattachedToActivityForConfigChanges(
+        activityPluginBinding: ActivityPluginBinding
+    ) {
+        activity = activityPluginBinding.activity
+        activityPluginBinding.addRequestPermissionsResultListener(this)
+    }
+
+    override fun onDetachedFromActivityForConfigChanges() {
+        activity = null
+    }
+
+    override fun onRequestPermissionsResult(
+        requestCode: Int,
+        permissions: Array<out String>,
+        grantResults: IntArray
+    ): Boolean {
+        if (requestCode != PERMISSION_REQUEST_CODE || grantResults.isEmpty()) return false
+        requestAccessCallback?.let { Handler(Looper.getMainLooper()).post(it) }
+        requestAccessCallback = null
+        return true
+    }
+
+    companion object {
+        private const val PERMISSION = Manifest.permission.WRITE_EXTERNAL_STORAGE
+        private val IMAGE_URI: Uri = MediaStore.Images.Media.EXTERNAL_CONTENT_URI
+        private val VIDEO_URI: Uri = MediaStore.Video.Media.EXTERNAL_CONTENT_URI
+        private const val PERMISSION_REQUEST_CODE = 1317298 // Anything unique in the app.
+        private val USE_EXTERNAL_STORAGE: Boolean = Build.VERSION.SDK_INT <= 29
+        private const val DEFAULT_TO_ALBUM = false
```

Comment:
For null saftey, this will not be used since the code you have in Dart package doesn't allow a null value but there is still a possibility of this to be null